src/server/common.cc

```diff
@@ -432,26 +432,22 @@ ThreadLocalMutex::~ThreadLocalMutex() {
 }
 
 void ThreadLocalMutex::lock() {
-  if (serialization_max_chunk_size != 0) {
-    DCHECK_EQ(EngineShard::tlocal(), shard_);
-    util::fb2::NoOpLock noop_lk_;
-    if (locked_fiber_ != nullptr) {
-      DCHECK(util::fb2::detail::FiberActive() != locked_fiber_);
-    }
-    cond_var_.wait(noop_lk_, [this]() { return !flag_; });
-    flag_ = true;
-    DCHECK_EQ(locked_fiber_, nullptr);
-    locked_fiber_ = util::fb2::detail::FiberActive();
+  DCHECK_EQ(EngineShard::tlocal(), shard_);
```

Comment:
I am not very much happy with this because we had a deadlock in production when I introduced this and we ended up disabling these locks for flows that do not preempt (when big value serialization is off). Now that we removed the `allow await` which controlled when a `write to sink` happens (which preempts the fiber and allows other fibers to interleave and write to journal) we must assume that all journal writes are preemptive and rely on this very lock for correctness.